src/components/UserSettings/GlobalSettings.js

```diff
@@ -189,6 +191,26 @@ class GlobalSettings extends Component {
           },
         ],
       },
+      advancedOptions: {
+        title: msg.advancedOptions(),
+        fields: [
+          {
+            title: msg.persistLanguage(),
+            tooltip: msg.persistLanguageTooltip(),
+            body: (<Switch
+              id={`${idPrefix}-persist-language`}
+              bsSize='normal'
+              title='normal'
+              value={draftValues.persistLocale}
+              onChange={(e, persist) => {
+                onChange('persistLocale')(persist)
+                if (persist === false) {
```

Comment:
This seems unnecessary to me. Please check how `notificationSnoozeDuration` is disabled  based on `showNotifications` flag.

Reply:
It's disable the `notificationSnoozeDuration` but it will also change the value in the backend, so IMO it's different case...